File: 203-remove-linked-list-elements/remove-linked-list-elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    /*
    let sentinel = new ListNode();
    sentinel.next = head;

    let prev = sentinel;
    while(prev&&prev.next){
        if(prev.next.val===val){
            prev.next=prev.next.next;
        }else{
            prev=prev.next;
        }
    }
    return sentinel.next;*/
// Remove matching nodes
    while(head && head.val===val){
        head = head.next;
    }
// now list is empty
if(head===null) return null;

// Remaning not removed elements
let curr = head;
while(curr&&curr.next){
    if(curr.next.val===val){
        curr.next=curr.next.next;
    }else{
        curr=curr.next;
    }
}
return head;
    
};

File: 203-remove-linked-list-elements/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {

// Remove matching nodes from the beginning
        while(head!=null && head.val==val){
           head = head.next;
        }
// if list becomes empty
if(head==null) return null;
// now head is either null or a valid starting node
    ListNode curr = head;

    while(curr!=null && curr.next!=null){
        if(curr.next.val==val){
            curr.next=curr.next.next;
        }else{
            curr=curr.next;
        }
    }
    return head;

        
    }
}